lib/msf/core/reflective_dll_loader.rb

```diff
@@ -48,17 +52,26 @@ def load_rdi_dll_from_data(dll_data)
 
   private
 
-  def parse_pe(dll)
+  def parse_pe(dll, loader_name: 'ReflectiveLoader', loader_ordinal: EXPORT_REFLECTIVELOADER)
     pe = Rex::PeParsey::Pe.new(Rex::ImageSource::Memory.new(dll))
     offset = nil
 
-    pe.exports.entries.each do |e|
-      if e.name =~ /^\S*ReflectiveLoader\S*/
-        offset = pe.rva_to_file_offset(e.rva)
-        break
+    unless loader_name.nil?
+      pe.exports.entries.each do |e|
+        if e.name =~ /^\S*${loader_name}\S*/
```

Comment:
I'm pretty sure this should be:

```suggestion
        if e.name =~ /^\S*#{loader_name}\S*/
```

Without that change my test case RDLL (which happens to use a ReflectiveLoader ordinal of 3) doesn't load.

Reply:
Sorry, I should have provided more context here.

My test involved getting a 64-bit Meterpreter session on a Windows 7 SP1 system, and then running the `post/windows/manage/reflective_dll_inject` module. For that module, I specified the `PID` of a running calc process and the `PATH` was set to [this test case](https://github.com/zeroSteiner/reflective-polymorphism/releases/download/v1.1/ReflectivePolymorphism.v1.1.x64.dll) DLL. That test case DLL is reflectively loadable and it exports the `ReflectiveLoader` name, but the ordinal is 3 and not 1. On successful injection, the calc process should spawn a message box saying "Select OK to proceed." showing that the DLL was correctly injected.

Reply:
Sorted!